Index: src/kml/base/contrib/minizip/unzip.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/kml/base/contrib/minizip/unzip.c	(revision 916a801ed3143ab82c07ec108bad271aa441da16)
+++ src/kml/base/contrib/minizip/unzip.c	(date 1595409052483)
@@ -184,7 +184,7 @@
 }
 
 #ifndef NOUNCRYPT
-#include <minizip/crypt.h>
+#include <crypt.h>
 #endif
 
 /* ===========================================================================
