Index: apps/cloud_composer/include/pcl/apps/cloud_composer/items/fpfh_item.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/cloud_composer/include/pcl/apps/cloud_composer/items/fpfh_item.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/cloud_composer/include/pcl/apps/cloud_composer/items/fpfh_item.h	(date 1588188138057)
@@ -43,7 +43,7 @@
 #include <pcl/apps/cloud_composer/items/cloud_composer_item.h>
 #include <pcl/visualization/pcl_plotter.h>
 
-class QVTKWidget;
+class QVTKOpenGLNativeWidget;
 
 namespace pcl
 {
@@ -74,7 +74,7 @@
         pcl::PointCloud<pcl::FPFHSignature33>::Ptr fpfh_ptr_;
         double radius_;
         pcl::visualization::PCLPlotter::Ptr plot_;
-        QVTKWidget *qvtk_;
+        QVTKOpenGLNativeWidget *qvtk_;
         QWidget *hist_page_;
     };
 
Index: tools/mesh_sampling.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- tools/mesh_sampling.cpp	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ tools/mesh_sampling.cpp	(date 1588136905672)
@@ -87,7 +87,7 @@
 
   double A[3], B[3], C[3];
   vtkIdType npts = 0;
-  vtkIdType *ptIds = nullptr;
+  const vtkIdType *ptIds = nullptr;
   polydata->GetCellPoints (el, npts, ptIds);
   polydata->GetPoint (ptIds[0], A);
   polydata->GetPoint (ptIds[1], B);
@@ -138,7 +138,8 @@
 
   double p1[3], p2[3], p3[3], totalArea = 0;
   std::vector<double> cumulativeAreas (cells->GetNumberOfCells (), 0);
-  vtkIdType npts = 0, *ptIds = nullptr;
+  vtkIdType npts = 0;
+  const vtkIdType *ptIds = nullptr;
   std::size_t cellId = 0;
   for (cells->InitTraversal (); cells->GetNextCell (npts, ptIds); cellId++)
   {
Index: apps/src/organized_segmentation_demo.ui
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/src/organized_segmentation_demo.ui	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/src/organized_segmentation_demo.ui	(date 1588195196457)
@@ -37,7 +37,7 @@
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="3">
-     <widget class="QVTKWidget" name="qvtk_widget">
+     <widget class="QVTKOpenGLNativeWidget" name="qvtk_widget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>255</horstretch>
@@ -214,9 +214,9 @@
  </widget>
  <customwidgets>
   <customwidget>
-   <class>QVTKWidget</class>
+   <class>QVTKOpenGLNativeWidget</class>
    <extends>QWidget</extends>
-   <header>QVTKWidget.h</header>
+   <header>QVTKOpenGLNativeWidget.h</header>
   </customwidget>
  </customwidgets>
  <resources>
Index: io/src/vtk_lib_io.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- io/src/vtk_lib_io.cpp	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ io/src/vtk_lib_io.cpp	(date 1587795717215)
@@ -343,7 +343,7 @@
 
   // Now handle the polygons
   mesh.polygons.resize (nr_polygons);
-  vtkIdType* cell_points;
+  const vtkIdType* cell_points;
   vtkIdType nr_cell_points;
   vtkCellArray * mesh_polygons = poly_data->GetPolys ();
   mesh_polygons->InitTraversal ();
Index: apps/cloud_composer/src/cloud_view.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/cloud_composer/src/cloud_view.cpp	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/cloud_composer/src/cloud_view.cpp	(date 1588188370900)
@@ -7,15 +7,15 @@
 
 #include <QDebug>
 
-#include <QVTKWidget.h>
+#include <vtk-9.0/QVTKOpenGLNativeWidget.h>
 
 pcl::cloud_composer::CloudView::CloudView (QWidget* parent)
   : QWidget (parent)
 {
   vis_.reset (new pcl::visualization::PCLVisualizer ("", false));
   vis_->getInteractorStyle ()->setKeyboardModifier (pcl::visualization::INTERACTOR_KB_MOD_SHIFT);
-  //Create the QVTKWidget
-  qvtk_ = new QVTKWidget (this);
+  //Create the QVTKOpenGLNativeWidget
+  qvtk_ = new QVTKOpenGLNativeWidget (this);
   qvtk_->SetRenderWindow (vis_->getRenderWindow ());
   initializeInteractorSwitch ();
   vis_->setupInteractor (qvtk_->GetInteractor (), qvtk_->GetRenderWindow (), style_switch_);
@@ -30,8 +30,8 @@
   model_ = model;
   vis_.reset (new pcl::visualization::PCLVisualizer ("", false));
  // vis_->getInteractorStyle ()->setKeyboardModifier (pcl::visualization::INTERACTOR_KB_MOD_SHIFT);
-  //Create the QVTKWidget
-  qvtk_ = new QVTKWidget (this);
+  //Create the QVTKOpenGLNativeWidget
+  qvtk_ = new QVTKOpenGLNativeWidget (this);
   qvtk_->SetRenderWindow (vis_->getRenderWindow ());
   initializeInteractorSwitch ();
   vis_->setupInteractor (qvtk_->GetInteractor (), qvtk_->GetRenderWindow (), style_switch_);
Index: apps/modeler/src/render_window.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/modeler/src/render_window.cpp	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/modeler/src/render_window.cpp	(date 1588195196470)
@@ -50,7 +50,7 @@
 
 //////////////////////////////////////////////////////////////////////////////////////////////
 pcl::modeler::RenderWindow::RenderWindow(RenderWindowItem* render_window_item, QWidget *parent, Qt::WindowFlags flags)
-  : QVTKWidget(parent, flags),
+  : QVTKOpenGLNativeWidget(parent, flags),
   axes_(vtkSmartPointer<vtkCubeAxesActor>::New()),
   render_window_item_(render_window_item)
 {
@@ -107,7 +107,7 @@
 {
   dynamic_cast<SceneTree*>(render_window_item_->treeWidget())->selectRenderWindowItem(render_window_item_);
 
-  QVTKWidget::focusInEvent(event);
+  QVTKOpenGLNativeWidget::focusInEvent(event);
 }
 
 //////////////////////////////////////////////////////////////////////////////////////////////
Index: visualization/include/pcl/visualization/point_cloud_geometry_handlers.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- visualization/include/pcl/visualization/point_cloud_geometry_handlers.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ visualization/include/pcl/visualization/point_cloud_geometry_handlers.h	(date 1587797414803)
@@ -45,6 +45,7 @@
 #include <pcl/point_cloud.h>
 #include <pcl/common/io.h>
 // VTK includes
+#include <vtkRendererCollection.h>
 #include <vtkSmartPointer.h>
 #include <vtkPoints.h>
 #include <vtkFloatArray.h>
Index: apps/cloud_composer/include/pcl/apps/cloud_composer/point_selectors/interactor_style_switch.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/cloud_composer/include/pcl/apps/cloud_composer/point_selectors/interactor_style_switch.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/cloud_composer/include/pcl/apps/cloud_composer/point_selectors/interactor_style_switch.h	(date 1588195196450)
@@ -45,7 +45,7 @@
 #include <pcl/visualization/common/ren_win_interact_map.h>
 #include <pcl/visualization/pcl_visualizer.h>
 
-class QVTKWidget;
+class QVTKOpenGLNativeWidget;
 
 namespace pcl
 {
@@ -93,7 +93,7 @@
         initializeInteractorStyles (pcl::visualization::PCLVisualizer::Ptr vis, ProjectModel* model);
         
         inline void 
-        setQVTKWidget (QVTKWidget* qvtk) { qvtk_ = qvtk; }
+        setQVTKWidget (QVTKOpenGLNativeWidget* qvtk) { qvtk_ = qvtk; }
                 
         void
         setCurrentInteractorStyle (interactor_styles::INTERACTOR_STYLES interactor_style);
@@ -134,7 +134,7 @@
         vtkSmartPointer<vtkPointPicker> point_picker_;
         
         /** \brief Internal pointer to QVTKWidget that this Switch works with */
-        QVTKWidget* qvtk_;
+        QVTKOpenGLNativeWidget* qvtk_;
         /** \brief Internal pointer to PCLVisualizer that this Switch works with */
         pcl::visualization::PCLVisualizer::Ptr vis_;
       private:
Index: apps/cloud_composer/src/items/fpfh_item.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/cloud_composer/src/items/fpfh_item.cpp	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/cloud_composer/src/items/fpfh_item.cpp	(date 1588187157431)
@@ -3,7 +3,7 @@
 
 #include <QGridLayout>
 
-#include <QVTKWidget.h>
+#include <QVTKOpenGLNativeWidget.h>
 
 pcl::cloud_composer::FPFHItem::FPFHItem (QString name, const pcl::PointCloud<pcl::FPFHSignature33>::Ptr& fpfh_ptr, double radius)
   : CloudComposerItem (std::move(name))
@@ -40,7 +40,7 @@
   if (!plot_)
   {
     plot_.reset (new pcl::visualization::PCLPlotter);
-    qvtk_ = new QVTKWidget ();
+    qvtk_ = new QVTKOpenGLNativeWidget ();
     hist_page_ = new QWidget ();
     QGridLayout *mainLayout = new QGridLayout (hist_page_);
     mainLayout-> addWidget (qvtk_,0,0);
@@ -49,8 +49,8 @@
   //Plot the histogram
   plot_->addFeatureHistogram (*fpfh_ptr_, fpfh_ptr_->width, data(ItemDataRole::ITEM_ID).toString().toStdString ());
   //Set the render window of the QVTK widget, update
-  plot_->setViewInteractor (vtkSmartPointer<vtkRenderWindowInteractor> (qvtk_->GetInteractor ()));
-  qvtk_->SetRenderWindow (plot_->getRenderWindow ());
+  plot_->setViewInteractor (vtkSmartPointer<vtkRenderWindowInteractor> (qvtk_->interactor ()));
+  qvtk_->setRenderWindow (plot_->getRenderWindow ());
   qvtk_->show ();
   qvtk_->update ();
   
Index: visualization/include/pcl/visualization/interactor_style.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- visualization/include/pcl/visualization/interactor_style.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ visualization/include/pcl/visualization/interactor_style.h	(date 1588136469589)
@@ -52,7 +52,7 @@
 #endif
 #include <vtkInteractorStyleRubberBandPick.h>
 
-class vtkRendererCollection;
+#include <vtk-9.0/vtkRendererCollection.h>
 class vtkLegendScaleActor;
 class vtkScalarBarActor;
 class vtkPNGWriter;
Index: visualization/include/pcl/visualization/vtk/vtkVertexBufferObjectMapper.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- visualization/include/pcl/visualization/vtk/vtkVertexBufferObjectMapper.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ visualization/include/pcl/visualization/vtk/vtkVertexBufferObjectMapper.h	(date 1588137310221)
@@ -25,13 +25,13 @@
 
 #include "vtkMapper.h"
 #include "vtkSmartPointer.h"
+#include <vtk-9.0/vtkShaderProgram.h>
 
 class vtkOpenGLRenderWindow;
 class vtkPolyData;
 class vtkRenderer;
 class vtkRenderWindow;
 class vtkShader2;
-class vtkShaderProgram2;
 class vtkVertexBufferObject;
 
 class PCL_EXPORTS vtkVertexBufferObjectMapper : public vtkMapper
@@ -56,7 +56,7 @@
   void SetInput(vtkDataSet *input);
   vtkPolyData *GetInput();
   
-  void SetProgram(vtkSmartPointer<vtkShaderProgram2> program)
+  void SetProgram(vtkSmartPointer<vtkShaderProgram> program)
   {
     this->program = program;
   }
@@ -118,7 +118,7 @@
   vtkVertexBufferObject *normalVbo;
 //  vtkVertexBufferObject *normalIndiceVbo;
 
-  vtkSmartPointer<vtkShaderProgram2> program;
+  vtkSmartPointer<vtkShaderProgram> program;
 
   int FillInputPortInformation(int, vtkInformation*) override;
 
Index: apps/src/render_views_tesselated_sphere.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/src/render_views_tesselated_sphere.cpp	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/src/render_views_tesselated_sphere.cpp	(date 1588137247925)
@@ -26,12 +26,14 @@
 #include <vtkRenderer.h>
 #include <vtkPolyDataMapper.h>
 #include <vtkPointPicker.h>
+#include <vtkIdTypeArray.h>
 
 void
 pcl::apps::RenderViewsTesselatedSphere::generateViews() {
   //center object
   double CoM[3];
-  vtkIdType npts_com = 0, *ptIds_com = nullptr;
+  vtkIdType npts_com = 0;
+  const vtkIdType *ptIds_com = nullptr;
   vtkSmartPointer<vtkCellArray> cells_com = polydata_->GetPolys ();
 
   double center[3], p1_com[3], p2_com[3], p3_com[3], totalArea_com = 0;
@@ -90,7 +92,8 @@
   // * Compute area of the mesh
   //////////////////////////////
   vtkSmartPointer<vtkCellArray> cells = mapper->GetInput ()->GetPolys ();
-  vtkIdType npts = 0, *ptIds = nullptr;
+  vtkIdType npts = 0;
+  const vtkIdType *ptIds = nullptr;
 
   double p1[3], p2[3], p3[3], totalArea = 0;
   for (cells->InitTraversal (); cells->GetNextCell (npts, ptIds);)
@@ -359,7 +362,8 @@
       polydata->BuildCells ();
 
       vtkSmartPointer<vtkCellArray> cells = polydata->GetPolys ();
-      vtkIdType npts = 0, *ptIds = nullptr;
+      vtkIdType npts = 0;
+      const vtkIdType *ptIds = nullptr;
 
       double p1[3], p2[3], p3[3], area, totalArea = 0;
       for (cells->InitTraversal (); cells->GetNextCell (npts, ptIds);)
Index: visualization/include/pcl/visualization/vtk/vtkVertexBufferObject.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- visualization/include/pcl/visualization/vtk/vtkVertexBufferObject.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ visualization/include/pcl/visualization/vtk/vtkVertexBufferObject.h	(date 1588135447841)
@@ -34,11 +34,12 @@
 #include "vtkgl.h" // Needed for gl data types exposed in API
 #include <pcl/pcl_macros.h>
 
+#include <vtk-9.0/vtkUnsignedCharArray.h>
+
 class vtkCellArray;
 class vtkDataArray;
 class vtkObject;
 class vtkPoints;
-class vtkUnsignedCharArray;
 class vtkOpenGLExtensionManager;
 class vtkRenderWindow;
 
Index: visualization/include/pcl/visualization/pcl_visualizer.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- visualization/include/pcl/visualization/pcl_visualizer.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ visualization/include/pcl/visualization/pcl_visualizer.h	(date 1588136575459)
@@ -53,10 +53,11 @@
 #include <pcl/visualization/area_picking_event.h>
 #include <pcl/visualization/interactor_style.h>
 
+#include <vtk-9.0/vtkRenderWindow.h>
+
 // VTK includes
 class vtkPolyData;
 class vtkTextActor;
-class vtkRenderWindow;
 class vtkOrientationMarkerWidget;
 class vtkAppendPolyData;
 class vtkRenderWindow;
Index: apps/src/openni_passthrough.ui
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/src/openni_passthrough.ui	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/src/openni_passthrough.ui	(date 1588195196467)
@@ -37,7 +37,7 @@
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
-     <widget class="QVTKWidget" name="qvtk_widget">
+     <widget class="QVTKOpenGLNativeWidget" name="qvtk_widget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>255</horstretch>
@@ -94,9 +94,9 @@
  </widget>
  <customwidgets>
   <customwidget>
-   <class>QVTKWidget</class>
+   <class>QVTKOpenGLNativeWidget</class>
    <extends>QWidget</extends>
-   <header>QVTKWidget.h</header>
+   <header>QVTKOpenGLNativeWidget.h</header>
   </customwidget>
  </customwidgets>
  <resources>
Index: surface/src/vtk_smoothing/vtk_utils.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- surface/src/vtk_smoothing/vtk_utils.cpp	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ surface/src/vtk_smoothing/vtk_utils.cpp	(date 1588136109400)
@@ -154,7 +154,7 @@
   }
 
   mesh.polygons.resize (nr_polygons);
-  vtkIdType* cell_points;
+  const vtkIdType* cell_points;
   vtkIdType nr_cell_points;
   vtkCellArray * mesh_polygons = poly_data->GetPolys ();
   mesh_polygons->InitTraversal ();
Index: apps/modeler/include/pcl/apps/modeler/points_actor_item.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/modeler/include/pcl/apps/modeler/points_actor_item.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/modeler/include/pcl/apps/modeler/points_actor_item.h	(date 1588137176625)
@@ -38,8 +38,7 @@
 
 #include <pcl/apps/modeler/channel_actor_item.h>
 #include <pcl/visualization/point_cloud_handlers.h>
-
-class vtkIdTypeArray;
+#include <vtk-9.0/vtkIdTypeArray.h>
 
 namespace pcl
 {
Index: apps/modeler/include/pcl/apps/modeler/render_window.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/modeler/include/pcl/apps/modeler/render_window.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/modeler/include/pcl/apps/modeler/render_window.h	(date 1588195196474)
@@ -38,7 +38,7 @@
 
 #include <vtkSmartPointer.h>
 
-#include <QVTKWidget.h>
+#include <vtk-9.0/QVTKOpenGLNativeWidget.h>
 
 class vtkCubeAxesActor;
 
@@ -48,7 +48,7 @@
   {
     class RenderWindowItem;
 
-    class RenderWindow : public QVTKWidget
+    class RenderWindow : public QVTKOpenGLNativeWidget
     {
       public:
         RenderWindow(RenderWindowItem* render_window_item, QWidget *parent = nullptr, Qt::WindowFlags flags = nullptr);
Index: apps/modeler/include/pcl/apps/modeler/surface_actor_item.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/modeler/include/pcl/apps/modeler/surface_actor_item.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/modeler/include/pcl/apps/modeler/surface_actor_item.h	(date 1588137133872)
@@ -38,8 +38,8 @@
 
 #include <pcl/apps/modeler/channel_actor_item.h>
 #include <pcl/visualization/point_cloud_handlers.h>
+#include <vtk-9.0/vtkIdTypeArray.h>
 
-class vtkIdTypeArray;
 
 namespace pcl
 {
Index: visualization/include/pcl/visualization/impl/pcl_visualizer.hpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- visualization/include/pcl/visualization/impl/pcl_visualizer.hpp	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ visualization/include/pcl/visualization/impl/pcl_visualizer.hpp	(date 1588136158083)
@@ -1699,18 +1699,16 @@
   {
     // Create polys from polyMesh.polygons
     vtkSmartPointer<vtkCellArray> cell_array = vtkSmartPointer<vtkCellArray>::New ();
-    vtkIdType *cell = cell_array->WritePointer (vertices.size (), vertices.size () * (max_size_of_polygon + 1));
+    cell_array->AllocateEstimate(vertices.size(), (max_size_of_polygon + 1));
     int idx = 0;
     if (!lookup.empty ())
     {
       for (std::size_t i = 0; i < vertices.size (); ++i, ++idx)
       {
         std::size_t n_points = vertices[i].vertices.size ();
-        *cell++ = n_points;
-        //cell_array->InsertNextCell (n_points);
+        cell_array->InsertNextCell (n_points);
         for (std::size_t j = 0; j < n_points; j++, ++idx)
-          *cell++ = lookup[vertices[i].vertices[j]];
-          //cell_array->InsertCellPoint (lookup[vertices[i].vertices[j]]);
+          cell_array->InsertCellPoint (lookup[vertices[i].vertices[j]]);
       }
     }
     else
@@ -1718,11 +1716,9 @@
       for (std::size_t i = 0; i < vertices.size (); ++i, ++idx)
       {
         std::size_t n_points = vertices[i].vertices.size ();
-        *cell++ = n_points;
-        //cell_array->InsertNextCell (n_points);
+        cell_array->InsertNextCell (n_points);
         for (std::size_t j = 0; j < n_points; j++, ++idx)
-          *cell++ = vertices[i].vertices[j];
-          //cell_array->InsertCellPoint (vertices[i].vertices[j]);
+          cell_array->InsertCellPoint (vertices[i].vertices[j]);
       }
     }
     vtkSmartPointer<vtkPolyData> polydata;
@@ -1876,16 +1872,16 @@
 
   // Update the cells
   cells = vtkSmartPointer<vtkCellArray>::New ();
-  vtkIdType *cell = cells->WritePointer (verts.size (), verts.size () * (max_size_of_polygon + 1));
+  cells->AllocateEstimate(verts.size(), (verts.size() * (max_size_of_polygon +1)));
   int idx = 0;
   if (!lookup.empty ())
   {
     for (std::size_t i = 0; i < verts.size (); ++i, ++idx)
     {
       std::size_t n_points = verts[i].vertices.size ();
-      *cell++ = n_points;
-      for (std::size_t j = 0; j < n_points; j++, cell++, ++idx)
-        *cell = lookup[verts[i].vertices[j]];
+      cells->InsertNextCell(n_points);
+      for (std::size_t j = 0; j < n_points; j++, ++idx)
+        cells->InsertCellPoint(lookup[verts[i].vertices[j]]);
     }
   }
   else
@@ -1893,9 +1889,9 @@
     for (std::size_t i = 0; i < verts.size (); ++i, ++idx)
     {
       std::size_t n_points = verts[i].vertices.size ();
-      *cell++ = n_points;
-      for (std::size_t j = 0; j < n_points; j++, cell++, ++idx)
-        *cell = verts[i].vertices[j];
+      cells->InsertNextCell(n_points);
+      for (std::size_t j = 0; j < n_points; j++, ++idx)
+        cells->InsertCellPoint(lookup[verts[i].vertices[j]]);
     }
   }
   cells->GetData ()->SetNumberOfValues (idx);
Index: apps/cloud_composer/include/pcl/apps/cloud_composer/cloud_view.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/cloud_composer/include/pcl/apps/cloud_composer/cloud_view.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/cloud_composer/include/pcl/apps/cloud_composer/cloud_view.h	(date 1588188647159)
@@ -43,6 +43,7 @@
 
 #include <pcl/visualization/pcl_visualizer.h>
 #include <pcl/apps/cloud_composer/point_selectors/interactor_style_switch.h>
+#include <vtk-9.0/QVTKOpenGLNativeWidget.h>
 
 class QItemSelection;
 class QStandardItem;
@@ -69,8 +70,8 @@
       setModel (ProjectModel* new_model);
       ProjectModel* 
       getModel () const { return model_; }
-      
-      QVTKWidget* 
+
+      QVTKOpenGLNativeWidget*
       getQVTK() const {return qvtk_; }
       
       pcl::visualization::PCLVisualizer::Ptr
@@ -141,7 +142,7 @@
       
       pcl::visualization::PCLVisualizer::Ptr vis_;
       ProjectModel* model_;
-      QVTKWidget* qvtk_;
+      QVTKOpenGLNativeWidget* qvtk_;
       vtkSmartPointer<InteractorStyleSwitch> style_switch_;
       
       vtkSmartPointer<vtkOrientationMarkerWidget> axes_widget_;
Index: apps/modeler/include/pcl/apps/modeler/normals_actor_item.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/modeler/include/pcl/apps/modeler/normals_actor_item.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/modeler/include/pcl/apps/modeler/normals_actor_item.h	(date 1588137152428)
@@ -38,8 +38,7 @@
 
 #include <pcl/apps/modeler/channel_actor_item.h>
 #include <pcl/visualization/point_cloud_handlers.h>
-
-class vtkIdTypeArray;
+#include <vtk-9.0/vtkIdTypeArray.h>
 
 namespace pcl
 {
Index: apps/src/manual_registration/manual_registration.ui
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/src/manual_registration/manual_registration.ui	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/src/manual_registration/manual_registration.ui	(date 1588195196460)
@@ -37,7 +37,7 @@
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="3">
-     <widget class="QVTKWidget" name="qvtk_widget_src">
+     <widget class="QVTKOpenGLNativeWidget" name="qvtk_widget_src">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>255</horstretch>
@@ -133,7 +133,7 @@
      </layout>
     </item>
     <item row="0" column="7">
-     <widget class="QVTKWidget" name="qvtk_widget_dst">
+     <widget class="QVTKOpenGLNativeWidget" name="qvtk_widget_dst">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>255</horstretch>
@@ -161,9 +161,9 @@
  </widget>
  <customwidgets>
   <customwidget>
-   <class>QVTKWidget</class>
+   <class>QVTKOpenGLNativeWidget</class>
    <extends>QWidget</extends>
-   <header>QVTKWidget.h</header>
+   <header>QVTKOpenGLNativeWidget.h</header>
   </customwidget>
  </customwidgets>
  <resources/>
Index: apps/src/pcd_video_player/pcd_video_player.ui
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/src/pcd_video_player/pcd_video_player.ui	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/src/pcd_video_player/pcd_video_player.ui	(date 1588188195680)
@@ -25,7 +25,7 @@
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
-      <widget class="QVTKWidget" name="qvtkWidget"/>
+      <widget class="QVTKOpenGLNativeWidget" name="qvtkWidget"/>
      </item>
      <item>
       <widget class="QSlider" name="indexSlider">
@@ -145,9 +145,9 @@
  </widget>
  <customwidgets>
   <customwidget>
-   <class>QVTKWidget</class>
+   <class>QVTKOpenGLNativeWidget</class>
    <extends>QWidget</extends>
-   <header>QVTKWidget.h</header>
+   <header>QVTKOpenGLNativeWidget.h</header>
   </customwidget>
  </customwidgets>
  <resources/>
Index: apps/src/pcd_video_player/pcd_video_player.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/src/pcd_video_player/pcd_video_player.cpp	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ apps/src/pcd_video_player/pcd_video_player.cpp	(date 1588188195666)
@@ -93,8 +93,8 @@
 
   // Set up the qvtk window
   vis_.reset (new pcl::visualization::PCLVisualizer ("", false));
-  ui_->qvtkWidget->SetRenderWindow (vis_->getRenderWindow ());
-  vis_->setupInteractor (ui_->qvtkWidget->GetInteractor (), ui_->qvtkWidget->GetRenderWindow ());
+  ui_->qvtkWidget->setRenderWindow (vis_->getRenderWindow ());
+  vis_->setupInteractor (ui_->qvtkWidget->interactor (), ui_->qvtkWidget->renderWindow ());
   vis_->getInteractorStyle ()->setKeyboardModifier (pcl::visualization::INTERACTOR_KB_MOD_SHIFT);
   ui_->qvtkWidget->update ();
 
Index: visualization/src/interactor_style.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- visualization/src/interactor_style.cpp	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ visualization/src/interactor_style.cpp	(date 1587797500180)
@@ -137,7 +137,7 @@
 {
   FindPokedRenderer (Interactor->GetEventPosition ()[0], Interactor->GetEventPosition ()[1]);
 
-  ofstream ofs_cam (file.c_str ());
+  std::ofstream ofs_cam (file.c_str ());
   if (!ofs_cam.is_open ())
   {
     return (false);
Index: visualization/src/pcl_visualizer.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- visualization/src/pcl_visualizer.cpp	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ visualization/src/pcl_visualizer.cpp	(date 1588136392583)
@@ -3159,16 +3159,16 @@
 
   // Update the cells
   cells = vtkSmartPointer<vtkCellArray>::New ();
-  vtkIdType *cell = cells->WritePointer (verts.size (), verts.size () * (max_size_of_polygon + 1));
+  cells->AllocateEstimate(verts.size (), verts.size () * (max_size_of_polygon + 1));
   int idx = 0;
   if (!lookup.empty ())
   {
     for (std::size_t i = 0; i < verts.size (); ++i, ++idx)
     {
       std::size_t n_points = verts[i].vertices.size ();
-      *cell++ = n_points;
-      for (std::size_t j = 0; j < n_points; j++, cell++, ++idx)
-        *cell = lookup[verts[i].vertices[j]];
+      cells->InsertNextCell(n_points);
+      for (std::size_t j = 0; j < n_points; j++, ++idx)
+        cells->InsertCellPoint(lookup[verts[i].vertices[j]]);
     }
   }
   else
@@ -3176,9 +3176,9 @@
     for (std::size_t i = 0; i < verts.size (); ++i, ++idx)
     {
       std::size_t n_points = verts[i].vertices.size ();
-      *cell++ = n_points;
-      for (std::size_t j = 0; j < n_points; j++, cell++, ++idx)
-        *cell = verts[i].vertices[j];
+      cells->InsertNextCell(n_points);
+      for (std::size_t j = 0; j < n_points; j++, ++idx)
+        cells->InsertCellPoint(verts[i].vertices[j]);
     }
   }
   cells->GetData ()->SetNumberOfValues (idx);
@@ -3551,7 +3551,8 @@
 
   //center object
   double CoM[3];
-  vtkIdType npts_com = 0, *ptIds_com = nullptr;
+  vtkIdType npts_com = 0;
+  const vtkIdType *ptIds_com = nullptr;
   vtkSmartPointer<vtkCellArray> cells_com = polydata->GetPolys ();
 
   double center[3], p1_com[3], p2_com[3], p3_com[3], totalArea_com = 0;
@@ -3610,7 +3611,8 @@
   // * Compute area of the mesh
   //////////////////////////////
   vtkSmartPointer<vtkCellArray> cells = mapper->GetInput ()->GetPolys ();
-  vtkIdType npts = 0, *ptIds = nullptr;
+  vtkIdType npts = 0;
+  const vtkIdType *ptIds = nullptr;
 
   double p1[3], p2[3], p3[3], totalArea = 0;
   for (cells->InitTraversal (); cells->GetNextCell (npts, ptIds);)
@@ -3829,7 +3831,8 @@
     polydata->BuildCells ();
 
     vtkSmartPointer<vtkCellArray> cells = polydata->GetPolys ();
-    vtkIdType npts = 0, *ptIds = nullptr;
+    vtkIdType npts = 0;
+    const vtkIdType *ptIds = nullptr;
 
     double p1[3], p2[3], p3[3], area, totalArea = 0;
     for (cells->InitTraversal (); cells->GetNextCell (npts, ptIds);)
Index: surface/include/pcl/surface/vtk_smoothing/vtk.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- surface/include/pcl/surface/vtk_smoothing/vtk.h	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ surface/include/pcl/surface/vtk_smoothing/vtk.h	(date 1588136029653)
@@ -43,5 +43,6 @@
 #  pragma GCC system_header 
 #endif
 
+#include <vtkUnsignedCharArray.h>
 #include <vtkPolyData.h>
 #include <vtkSmartPointer.h>
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- CMakeLists.txt	(revision 6cd964569eaebdf193f8279616af7f5202aad9f8)
+++ CMakeLists.txt	(date 1588184348057)
@@ -367,25 +367,26 @@
 option(WITH_VTK "Build VTK-Visualizations" TRUE)
 if(WITH_VTK AND NOT ANDROID)
   set(PCL_VTK_COMPONENTS
-    vtkChartsCore
-    vtkCommonCore
-    vtkCommonDataModel
-    vtkCommonExecutionModel
-    vtkFiltersCore
-    vtkFiltersExtraction
-    vtkFiltersModeling
-    vtkImagingCore
-    vtkImagingSources
-    vtkInteractionStyle
-    vtkInteractionWidgets
-    vtkIOCore
-    vtkIOGeometry
-    vtkIOImage
-    vtkIOLegacy
-    vtkIOPLY
-    vtkRenderingAnnotation
-    vtkRenderingLOD
-    vtkViewsContext2D
+    ChartsCore
+    CommonCore
+    CommonDataModel
+    CommonExecutionModel
+    FiltersCore
+    FiltersExtraction
+    FiltersGeometry
+    FiltersModeling
+    ImagingCore
+    ImagingSources
+    InteractionStyle
+    InteractionWidgets
+    IOCore
+    IOGeometry
+    IOImage
+    IOLegacy
+    IOPLY
+    RenderingAnnotation
+    RenderingLOD
+    ViewsContext2D
   )
   find_package(VTK COMPONENTS ${PCL_VTK_COMPONENTS})
   if(VTK_FOUND AND ("${VTK_VERSION}" VERSION_LESS 6.2))
@@ -395,16 +396,15 @@
 
   if(VTK_FOUND)
     if(NOT DEFINED VTK_RENDERING_BACKEND)
-      # On old VTK versions this variable does not exist. In this case it is
-      # safe to assume OpenGL backend
-      set(VTK_RENDERING_BACKEND "OpenGL")
+      # In this case it is safe to assume OpenGL backend
+      set(VTK_RENDERING_BACKEND "OpenGL2")
     endif()
-    list(APPEND PCL_VTK_COMPONENTS vtkRenderingContext${VTK_RENDERING_BACKEND})
-
+    list(APPEND PCL_VTK_COMPONENTS RenderingContext${VTK_RENDERING_BACKEND})
+    message("${VTK_AVAILABLE_COMPONENTS}")
     if(WITH_QT)
-      if(";${VTK_MODULES_ENABLED};" MATCHES ";vtkGUISupportQt;" AND ";${VTK_MODULES_ENABLED};" MATCHES ";vtkRenderingQt;")
+      if(";${VTK_AVAILABLE_COMPONENTS};" MATCHES ";GUISupportQt;" AND ";${VTK_AVAILABLE_COMPONENTS};" MATCHES ";RenderingQt;")
         set(QVTK_FOUND ON)
-        list(APPEND PCL_VTK_COMPONENTS vtkRenderingQt vtkGUISupportQt)
+        list(APPEND PCL_VTK_COMPONENTS RenderingQt GUISupportQt)
       else()
         unset(QVTK_FOUND)
       endif()
