Subject: [PATCH] include-loki-headers-relpath
---
Index: core/XdmfItem.hpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/XdmfItem.hpp b/core/XdmfItem.hpp
--- core/XdmfItem.hpp	(revision 04a84bab0eb1568e0f1a27c8fb60c6931efda003)
+++ core/XdmfItem.hpp	(date 1668918952718)
@@ -36,7 +36,7 @@
 class XdmfVisitor;
 
 // Includes
-#include <loki/Visitor.h>
+#include "loki/Visitor.h"
 #include <libxml/xmlexports.h>
 #include <libxml/tree.h>
 #include <libxml/uri.h>
Index: core/XdmfVisitor.hpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/XdmfVisitor.hpp b/core/XdmfVisitor.hpp
--- core/XdmfVisitor.hpp	(revision 04a84bab0eb1568e0f1a27c8fb60c6931efda003)
+++ core/XdmfVisitor.hpp	(date 1668918952738)
@@ -33,7 +33,7 @@
 class XdmfItem;
 
 // Includes
-#include <loki/Visitor.h>
+#include "loki/Visitor.h"
 
 /**
  * @brief Perform an operation on an Xdmf tree structure.
Index: core/loki/Visitor.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/loki/Visitor.h b/core/loki/Visitor.h
--- core/loki/Visitor.h	(revision 04a84bab0eb1568e0f1a27c8fb60c6931efda003)
+++ core/loki/Visitor.h	(date 1668919151936)
@@ -22,7 +22,7 @@
 
 #include "Typelist.h"
 #include "HierarchyGenerators.h"
-#include "XdmfSharedPtr.hpp"
+#include <xdmf/XdmfSharedPtr.hpp>
 
 namespace Loki
 {
