--- src/core/Error.cpp	(revision a077afb396685124bddbe97713e9a68efbe4724a)
+++ src/core/Error.cpp	(date 1662800490235)
@@ -27,18 +27,18 @@
     #include <sstream>
     #include <windows.h>
 #endif
-
+#include <string.h>
 #include <Error.hpp>
 
 std::string getErrorString(int errnum)
 {
     char buffer[512];
 
-    if (errnum < sys_nerr)
+    try
     {
-        (void)std::snprintf(buffer, sizeof(buffer), "%s", sys_errlist[errnum]);
+        (void)std::snprintf(buffer, sizeof(buffer), "%s", strerror(errnum));
     }
-    else
+    catch (std::exception e)
     {
         (void)std::snprintf(buffer, sizeof(buffer), "Unknown error %d", errnum);
     }
