Index: simgear/scene/util/SGImageUtils.cxx
<+>UTF-8
====================================================================
--- simgear/scene/util/SGImageUtils.cxx	(revision a157e503020611a3e0caa1a450a01a97614c896a)
+++ simgear/scene/util/SGImageUtils.cxx	(date 1630748093991)
@@ -26,7 +26,8 @@
 #include <osg/Node>
 #include <osgDB/Options>
 #include <osgDB/Registry>
-
+#include <osgEarth/Registry>
+#include <osgEarth/Capabilities>
 #define LC "[ImageUtils] "
 
 
@@ -1037,7 +1038,7 @@
     if (!image)
         return false;
 
-//    const Capabilities& caps = Registry::capabilities();
+const osgEarth::Capabilities& caps = osgEarth::Registry::capabilities();
 
 #if !defined(OSG_GLES2_AVAILABLE) && !defined(OSG_GLES3_AVAILABLE)
 
Index: simgear/scene/dem/SGDemTile_gdal.cxx
<+>UTF-8
===================================================================
--- simgear/scene/dem/SGDemTile_gdal.cxx	(revision a157e503020611a3e0caa1a450a01a97614c896a)
+++ simgear/scene/dem/SGDemTile_gdal.cxx	(date 1630748179727)
@@ -172,7 +172,7 @@
                      && (pszMethod == NULL || EQUALN(pszMethod,"GCP_",4)) ) {
                 pszThisSourceSRS = GDALGetGCPProjection( hSrcDS );
             } else if( pszMethod != NULL && EQUAL(pszMethod,"RPC") ) {
-                pszThisSourceSRS = SRS_WKT_WGS84;
+                pszThisSourceSRS = SRS_DN_WGS84;
             } else {
                 pszThisSourceSRS = "";
             }
Index: simgear/CMakeLists.txt
<+>UTF-8
===================================================================
--- simgear/CMakeLists.txt	(revision a157e503020611a3e0caa1a450a01a97614c896a)
+++ simgear/CMakeLists.txt	(date 1630749694684)
@@ -188,6 +188,11 @@
         ${OPENSCENEGRAPH_LIBRARIES}
     )
 
+    target_link_libraries(SimGearScene PUBLIC
+            SimGearCore
+            ${OSGEARTH_LIBRARY}
+            )
+
     target_link_libraries(SimGearScene PRIVATE
         ${ZLIB_LIBRARY}
         ${OPENAL_LIBRARY}
Index: CMakeLists.txt
<+>UTF-8
===================================================================
--- CMakeLists.txt	(revision a157e503020611a3e0caa1a450a01a97614c896a)
+++ CMakeLists.txt	(date 1630749365606)
@@ -261,7 +261,7 @@
     endif(ENABLE_SOUND)
 
     find_package(OpenSceneGraph 3.2.0 REQUIRED osgText osgSim osgDB osgParticle osgGA osgViewer osgUtil)
-
+    find_package(OsgEarth REQUIRED)
     if (MSVC)
       set(CMAKE_REQUIRED_INCLUDES ${OPENSCENEGRAPH_INCLUDE_DIRS})
       # ensure OSG was compiled with OSG_USE_UTF8_FILENAME set
