---
Index: src/Sawyer/CommandLine.C
<+>UTF-8
===================================================================
--- src/Sawyer/CommandLine.C	(revision 7bf80a2f885a71d82d1bac30b71b3dcc3532ded9)
+++ src/Sawyer/CommandLine.C	(date 1780112333742)
@@ -41,7 +41,7 @@
 # if defined(__APPLE__) && defined(__MACH__)
 #  include <termios.h>
 # else
-#  include <termio.h>
+#  include <termios.h>
 # endif
 #include <sys/ioctl.h>
 #endif
