--- CMakeLists.txt
+++ CMakeLists.txt
@@ -61,9 +61,9 @@
 add_executable(3dfier ${SRC_FILES})
 set_target_properties(
   3dfier
-  PROPERTIES CXX_STANDARD 11
+  PROPERTIES CXX_STANDARD 14
 )
 
 target_link_libraries( 3dfier ${CGAL_LIBRARIES} ${CGAL_3RD_PARTY_LIBRARIES} ${GDAL_LIBRARY} ${YAML_CPP_LIBRARIES} Boost::program_options Boost::filesystem Boost::locale Boost::chrono LASlib)
 
-install(TARGETS 3dfier DESTINATION bin)
\ No newline at end of file
+install(TARGETS 3dfier DESTINATION bin)
