Index: plugins/openscenegraph/io/OSGReader.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- a/plugins/openscenegraph/io/OSGReader.cpp	(revision 766aaa286ba8fd13384ef159f16a75bd8b8cfdb7)
+++ plugins/openscenegraph/io/OSGReader.cpp	(date 1574523217930)
@@ -52,7 +52,7 @@
     }
 #ifdef OSG_VERSION_GREATER_THAN
 #if OSG_VERSION_GREATER_THAN(3,2,0)
-    inline void operator () (const osg::Vec3& v1, const osg::Vec3& v2, const osg::Vec3& v3, bool treatVertexDataAsTemporary)
+    inline void operator () (const osg::Vec3& v1, const osg::Vec3& v2, const osg::Vec3& v3)
 #else
     inline void operator () (const osg::Vec3& v1, const osg::Vec3& v2, const osg::Vec3& v3)
 #endif
