Index: Remote/PoissonReconstruction.remote.cmake
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Remote/PoissonReconstruction.remote.cmake	(revision e6d8ac87a270e4f867fe7785fd11a8296d864cc4)
+++ Remote/PoissonReconstruction.remote.cmake	(date 1592186883805)
@@ -3,7 +3,7 @@
 #
 vtk_fetch_module(PoissonReconstruction
   "Poisson Surface reconstruction from unorganized points"
-  GIT_REPOSITORY https://github.com/lorensen/PoissonReconstruction
-  # 2019 Feb 13
-  GIT_TAG 3b23f5a4aa1521b8f817682add048342cd3fd305
+  GIT_REPOSITORY https://github.com/nickmcummins/PoissonReconstruction
+  # 2020 Jun 14
+  GIT_TAG a9b093637884626c612f3b3d8bd6e622eec467ee
   )
