Index: src/saga_core/saga_gui/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/saga_core/saga_gui/CMakeLists.txt	(revision a95ee9852ab761a1514c2886795b247f0f260b04)
+++ src/saga_core/saga_gui/CMakeLists.txt	(date 1578201757718)
@@ -6,6 +6,7 @@
 	active.cpp
 	active_attributes.cpp
 	active_attributes_control.cpp
+	active_info.cpp
 	active_description.cpp
 	active_history.cpp
 	active_legend.cpp
@@ -102,6 +103,7 @@
 	active.h
 	active_attributes.h
 	active_attributes_control.h
+	active_info.h
 	active_description.h
 	active_history.h
 	active_legend.h
Index: src/tools/grid/grid_gridding/nn/delaunay.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/saga_core/saga_gui/saga.cpp	(revision a95ee9852ab761a1514c2886795b247f0f260b04)
+++ src/saga_core/saga_gui/saga.cpp	(date 1578201783052)
@@ -133,7 +133,7 @@
 	 * floating point values. wxXLocale is currently not fully
 	 * implemented (wxPrintf_l and similar still missing). */
 	//setlocale(LC_NUMERIC, "C");
-	m_wxLocale.Init(wxLANGUAGE_ENGLISH);
+	m_wxLocale.Init(wxLANGUAGE_ENGLISH_US);
 
 	_Init_Config();
 
Index: src/tools/projection/pj_proj4/crs_base.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/tools/projection/pj_proj4/crs_transform.cpp	(revision 95038b874d6d5f00328ecc385d365874496f53d2)
+++ src/tools/projection/pj_proj4/crs_transform.cpp	(date 1592189081557)
@@ -58,7 +58,7 @@
 //---------------------------------------------------------
 #ifndef PROJ6
 	extern "C" {
-		#include <projects.h>
+		#include <vtk-9.0/vtklibproj/src/projects.h>
 	}

 	#define PROJ4_FREE(p)	if( p )	{	pj_free((PJ *)p);	p	= NULL;	}
Index: saga-gis/src/tools/projection/pj_proj4/crs_base.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/tools/projection/pj_proj4/crs_base.cpp	(revision 95038b874d6d5f00328ecc385d365874496f53d2)
+++ src/tools/projection/pj_proj4/crs_base.cpp	(date 1592189049091)
@@ -58,7 +58,7 @@
 //---------------------------------------------------------
 #ifndef PROJ6
 	extern "C" {
-		#include <projects.h>
+		#include <vtk-9.0/vtklibproj/src/projects.h>
 	}

 	#define PJ_GET_PROJS	pj_get_list_ref()
Index: src/tools/io/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/tools/io/CMakeLists.txt	(revision 95038b874d6d5f00328ecc385d365874496f53d2)
+++ src/tools/io/CMakeLists.txt	(date 1592193759809)
@@ -7,7 +7,7 @@
 if(WITH_GUI)
 	add_subdirectory(io_grid_image)
 endif()
-#add_subdirectory(io_pdal)
+add_subdirectory(io_pdal)
 add_subdirectory(io_shapes)
 add_subdirectory(io_shapes_dxf)
 #add_subdirectory(io_shapes_las)
