---
Index: .gitmodules
<+>UTF-8
===================================================================
diff --git a/.gitmodules b/.gitmodules
--- .gitmodules	(revision ff7926cd9462a2570c8c9863bd4cb0f06358342f)
+++ .gitmodules	(date 1746726995692)
@@ -35,7 +35,7 @@
 	url = https://github.com/ColinDuquesnoy/QDarkStyleSheet
 [submodule "plugins/core/Standard/qVoxFall"]
 	path = plugins/core/Standard/qVoxFall
-	url = ../../farmakis/qVoxFall.git
+	url = https://github.com/farmakis/qVoxFall.git
 [submodule "plugins/core/Standard/qG3Point"]
 	path = plugins/core/Standard/qG3Point
 	url = https://github.com/p-leroy/qG3Point
